File: dagger-java/app/src/main/java/com/coder/dagger_java/di/component/ActivityComponent.java
package com.coder.dagger_java.di.component;

import com.coder.dagger_java.MainActivity;
import com.coder.dagger_java.di.PerActivity;
import com.coder.dagger_java.di.module.ActivityModule;

import dagger.Component;

@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {

    void inject(MainActivity mainActivity);
}

File: dagger-kotlin/app/src/main/java/com/coder/dagger_kotlin/di/component/ActivityComponent.kt
package com.coder.dagger_kotlin.di.component

import com.coder.dagger_kotlin.MainActivity
import com.coder.dagger_kotlin.di.PerActivity
import com.coder.dagger_kotlin.di.module.ActivityModule
import dagger.Component

@PerActivity
@Component(dependencies = [ApplicationComponent::class], modules = [ActivityModule::class])
interface ActivityComponent {

    fun inject(mainActivity: MainActivity)
}